### PR TITLE
feat: add request id middleware

### PR DESCRIPTION
**Pull request by Haswf**
#### What type of PR is this?

feat: add a middleware to generate unique id for each request. Also see https://github.com/cloudwego/hertz/issues/156.

#### What this PR does / why we need it (English/Chinese):

en: add request id middleware
zh: 增加请求ID中间件

### CODE REVIEW

File: .licenserc.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+header:
+  license:
+    spdx-id: Apache-2.0
+    copyright-owner: CloudWeGo Authors
+    content: |
+      The MIT License (MIT)
```
Comment (Duslia) [inline .licenserc.yaml]: 不是加 mit license，cloudwego 都是 apache 的 license

File: example/customheader/main.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+func main() {
+	h := server.Default()
+
+	h.Use(
+		requestid.New(requestid.WithCustomHeaderStrKey("X-Trace-Id")),
+	)
+
+	// Example ping request.
+	h.GET("/ping", func(ctx context.Context, c *app.RequestContext) {
+		c.JSON(consts.StatusOK, utils.H{"ping": "pong"})
```
Comment (Duslia) [inline example/customheader/main.go:39]: 例子中可以说一下怎么取 uuid 吗？

Comment (Haswf) [reply example/customheader/main.go:39]: quickstart includes an example of how to retrieve request id form original request. Does that sounds good to you?

File: .licenserc.yaml (hunk 1 above)
Comment (Haswf) [reply .licenserc.yaml]: Fixed

File: README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# RequestID (This is a community driven project)
+ 
+Request ID middleware for Hertz framework, inspired by [requestid](https://github.com/gin-contrib/requestid), a Request ID middleware for Gin Framework.
```
Comment (Duslia) [inline README.md]: 后面不是给 gin 的，最后那句话是不是删了就可以了

File: README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+This project would not have been possible without the support from the CloudWeGo community and previous work done by the gin community.
+
+- Adds an identifier to the response using the `X-Request-ID` header. 
+- Passes the `X-Request-ID` value back to the caller if it's sent in the request headers.
+
+
+## Install
+```shell
+go get https://github.com/hertz-contrib/requestid
```
Comment (Duslia) [inline README.md]: 这个不对吧。安装是不是  go get github.com/hertz-contrib/requestid

File: example/customgenerator/main.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	h.Use(
+		// define your own request id generator here
+		requestid.New(requestid.WithGenerator(func() string {
+			return "cloudwego.io"
+		})),
+	)
+
+	// Example ping request.
+	h.GET("/ping", func(ctx context.Context, c *app.RequestContext) {
+		c.JSON(consts.StatusOK, utils.H{"ping": "pong"})
```
Comment (Duslia) [inline example/customgenerator/main.go]: 这个例子返回获取到的 requestid 吧

**Review by Duslia — APPROVED** [review]

**Pull request merged**